[CI] GenerativeIT test failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #8784 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/de6l7xdkzaaxs)
- [elasticsearch-periodic #8769 / openjdk23_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/cgtnpgnfktol4)
- [elasticsearch-pull-request #75524 / part-3](https://gradle-enterprise.elastic.co/s/vpo3zc7e7cbiw)
- [elasticsearch-pull-request #75225 / part-3](https://gradle-enterprise.elastic.co/s/4qlqpbw43olya)
- [elasticsearch-pull-request #75200 / part-3](https://gradle-enterprise.elastic.co/s/2vccry52gnfr6)
- [elasticsearch-pull-request #74788 / part-3](https://gradle-enterprise.elastic.co/s/ipx24v6aulbme)
- [elasticsearch-pull-request #74706 / part-3](https://gradle-enterprise.elastic.co/s/xd3mt3zcktrm2)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT.test" -Dtests.seed=AE2763A306A2F83 -Dtests.locale=sq-Latn-AL -Dtests.timezone=Atlantic/South_Georgia -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from sample_data_str,app_logs,fi* | sort service_id, client_ip DESC, destination.IP NULLS FIRST | limit 9425 | eval  xJRwHRezpds = -1299206307, ZuPpUkXhu = -1360937430036226416, `source.IP` = "IizPcFyV" | limit 9865 | keep xJRwHRezpds, `source.IP` | limit 14417 | rename `source.IP` AS `dVxoyvSH`| rename dVxoyvSH as message | lookup join message_types_lookup on message | drop xJRwHRezpds, *Rezpds | limit 13955 | mv_expand message | dissect message "%{RQZzvsuaWfI}" | rename RQZzvsuaWfI AS `message`, `message` AS FJaUmggBPj | dissect FJaUmggBPj "%{FJaUmggBPj} %{FJaUmggBPj}" | rename FJaUmggBPj AS `FvgUVtLZjcGv` | grok FvgUVtLZjcGv "%{WORD:FvgUVtLZjcGv}" | eval  FvgUVtLZjcGv = -2012120826, DRfKHTdnx = 1146940792054680116, FvgUVtLZjcGv = null, FvgUVtLZjcGv = null | keep *
error: Expecting at most [2] columns, got [3]
```

**Issue Reasons:**
- [main] 7 failures in test test (0.7% fail rate in 978 executions)
- [main] 5 failures in step part-3 (1.2% fail rate in 411 executions)
- [main] 5 failures in pipeline elasticsearch-pull-request (1.2% fail rate in 410 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -565,9 +565,6 @@ tests:
 - class: org.elasticsearch.search.query.VectorIT
   method: testFilteredQueryStrategy
   issue: https://github.com/elastic/elasticsearch/issues/129517
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/129453
 - class: org.elasticsearch.test.apmintegration.TracesApmIT
   method: testApmIntegration
   issue: https://github.com/elastic/elasticsearch/issues/129651

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java
@@ -13,7 +13,6 @@ import org.elasticsearch.xpack.esql.qa.rest.generative.command.pipe.DissectGener
 import org.elasticsearch.xpack.esql.qa.rest.generative.command.pipe.DropGenerator;
 import org.elasticsearch.xpack.esql.qa.rest.generative.command.pipe.EnrichGenerator;
 import org.elasticsearch.xpack.esql.qa.rest.generative.command.pipe.EvalGenerator;
-import org.elasticsearch.xpack.esql.qa.rest.generative.command.pipe.ForkGenerator;
 import org.elasticsearch.xpack.esql.qa.rest.generative.command.pipe.GrokGenerator;
 import org.elasticsearch.xpack.esql.qa.rest.generative.command.pipe.KeepGenerator;
 import org.elasticsearch.xpack.esql.qa.rest.generative.command.pipe.LimitGenerator;
@@ -54,7 +53,8 @@ public class EsqlQueryGenerator {
         DropGenerator.INSTANCE,
         EnrichGenerator.INSTANCE,
         EvalGenerator.INSTANCE,
-        ForkGenerator.INSTANCE,
+        // Awaits fix: https://github.com/elastic/elasticsearch/issues/129715
+        // ForkGenerator.INSTANCE,
         GrokGenerator.INSTANCE,
         KeepGenerator.INSTANCE,
         LimitGenerator.INSTANCE,

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/command/pipe/LimitGenerator.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/command/pipe/LimitGenerator.java
@@ -42,15 +42,9 @@ public class LimitGenerator implements CommandGenerator {
         List<List<Object>> output
     ) {
         int limit = (int) commandDescription.context().get(LIMIT);
-        boolean defaultLimit = false;
-        for (CommandDescription previousCommand : previousCommands) {
-            if (previousCommand.commandName().equals(LIMIT)) {
-                defaultLimit = true;
-            }
-        }
 
-        if (previousOutput.size() > limit && output.size() != limit || defaultLimit && previousOutput.size() < output.size()) {
-            return new ValidationResult(false, "Expecting [" + limit + "] records, got [" + output.size() + "]");
+        if (output.size() > limit) {
+            return new ValidationResult(false, "Expecting at most [" + limit + "] records, got [" + output.size() + "]");
         }
         return CommandGenerator.expectSameColumns(previousColumns, columns);
     }

